### ISSUE TITLE
[PEP 517] Relax installation-time "pep517<0.7.0" requirement

### ISSUE THREAD

**Issue by leycec** [kivy/python-for-android#2573]
`python-for-android` currently [requires obsolete versions of the `pep517` Python package at `setuptools`-based installation time via a `pep517<0.7.0` requirement in the top-level `setup.py` script](https://github.com/kivy/python-for-android/blob/afd9942a5afb9d3609ce7a01a387f0df492ce1b9/setup.py#L25). This is bad for any number of obvious reasons, including:

* `pep517 0.6.0`, the most recent version of this package to satisfy this constraint, is **[nearly three years old](https://pypi.org/project/pep517/0.6.0/#files)**. In Python years, `pep517` is the geriatric centenarian with insufferable incontinence that habitually screeches "Gituffmuhlaaaaawn!" while throwing mouldy wonder bread at passing codebases.
* `pep517 0.6.0` is so old that it's only tested to work under Python 3.4–3.7. Of those, only Python 3.7 has yet to hit its official End-of-Life (EOL) – *[but even that will happen shortly](https://endoflife.date/python).*
* `pep517 0.6.0` is no longer packaged by any system package managers (e.g., Debian's `apt`, Arch's `pacman`, Gentoo's `emerge`). It's unclear whether `pep517 0.6.0` even *can* be packaged by any system package managers or run under actively developed Python versions. By transitivity, **`python-for-android` itself can no longer be reasonably packaged for any Linux distros.**

In short, the `pep517<0.7.0` constraint needs to be relaxed to cover [the most recent stable release of `pep517`](https://pypi.org/project/pep517). In theory, this *should* be trivial; the p4a codebase only [internally imports from `pep517` twice in the `python-for-android.pythonpackage` submodule](https://github.com/kivy/python-for-android/blob/8336cea09c572c2a234ddda211226a1eb8b2ad1d/pythonforandroid/pythonpackage.py#L51).

Thanks as always for all the phenomenal volunteerism, p4a team! You make the impossible at least slightly possible. :cowboy_hat_face:

**Comment by leycec** [comment]
Thanks so much, @misl6. Kivy devs the best devs. :smiling_face_with_three_hearts:
